### PR TITLE
add debug message when something bad happened

### PR DESCRIPTION
**Pull request by xinjiacs**

### CODE REVIEW

**Comment by nshi** [comment]
This probably shouldn't go onto master because it records every single message the SpScheduler sees. Can you ask Phil to run the system test on your branch?

File: src/frontend/org/voltdb/iv2/SpScheduler.java [hunk 1]
```diff
@@ -1285,15 +1286,28 @@ private long getRepairLogTruncationHandleForReplicas()
...
+        m_ll.add(new RuntimeException("Updating truncation point from " +
+                TxnEgo.txnIdToString(m_repairLogTruncationHandle) +
+                "to" + TxnEgo.txnIdToString(newHandle)));
+
         assert newHandle >= m_repairLogTruncationHandle : "new handle: " + newHandle + ", repairLog:" + m_repairLogTruncationHandle;
+
         if (newHandle > m_repairLogTruncationHandle) {
             m_repairLogTruncationHandle = newHandle;
 
-            if (! m_isLeader) {
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: You should not remove this line. It keeps the replicas' truncation point static until the leader tells them to update.

Comment (xinjiacs) [inline src/frontend/org/voltdb/iv2/SpScheduler.java]: Should we move this IF check to the beginning of this function? The replica should not even update itself as well, only waiting leader messages. @nshi

**Comment by xinjiacs** [comment]
Leaving the debug messages on branch.

**Pull request closed without merging**